pkg/foundation/database/badger/db.go

```diff
@@ -47,6 +48,15 @@ func New(l zerolog.Logger, path string) (*DB, error) {
 	}, nil
 }
 
+func (d *DB) Ping(ctx context.Context) error {
+	key := "test-" + uuid.NewString()
+	defer d.Set(ctx, key, nil) //nolint:errcheck // clean-up
+
+	// todo ideally, we find a way to test the connection
+	// without inserting data
```

Comment:
There is no connection, badger is embedded in the process and flushes data to the file system from time to time.

I tried running Conduit and deleting `conduit.db` while it was running. The health check was saying everything is fine, and indeed the service was running fine, even mutating and running pipelines was still possible. But once I wanted to stop Conduit it threw an error saying that it's unable to open `conduit.db/...`. I tried changing the health check to execute `d.db.Sync()` and not even that threw an error, Conduit was "healthy" and proceeded to normally work until it was stopped.

Long story short - I think apart from checking if the db is closed there's not much we can check here 😕

Reply:
> There is no connection, badger is embedded in the process and flushes data to the file system from time to time.

Yup, yup, I didn't mean a "permanent" connection, as we have it in DBs, but more like a "connection" between BadgerDB and the file system. 

Besides a method in BadgerDB's StreamWriter (which also deletes all the data:) ) I didn't find a good way to do this check. The only thing which we could potentially do is to check if the dir is there and if we can list the contents.